### PR TITLE
#200 Restructure http endpoints according to latest design

### PR DESCRIPTION
**Pull request by diehuxx**
Addresses #200 
New design here https://github.com/TBD54566975/tbdex/issues/225#issuecomment-1986057713

### CODE REVIEW

File: httpclient/src/main/kotlin/tbdex/sdk/httpclient/TbdexHttpClient.kt [hunk 1]
```diff
@@ -102,7 +90,18 @@ object TbdexHttpClient {
     val body: RequestBody = Json.stringify(CreateExchangeRequest(rfq))
```
Comment (KendallWeihe) [inline httpclient/src/main/kotlin/tbdex/sdk/httpclient/TbdexHttpClient.kt:90]: The line above this (it's out of my GitHub diff view here so having to comment here)

```kt
val path = "/exchanges/$exchangeId"
```

Should just be `/exchanges` right?

**Review by KendallWeihe — APPROVED** [review]
One comment left with regards to the create exchange path, and build is failing

Other than that this LGTM

File: httpclient/src/main/kotlin/tbdex/sdk/httpclient/TbdexHttpClient.kt (hunk 1 above)
Comment (diehuxx) [reply httpclient/src/main/kotlin/tbdex/sdk/httpclient/TbdexHttpClient.kt:86]: fixed

File: httpserver/src/main/kotlin/tbdex/sdk/httpserver/handlers/SubmitMessage.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * Handles the submission of a close message by parsing the incoming message,
+ * validating the submission, and executing the specified callback if provided.
+ *
+ * @param call The [ApplicationCall] instance representing the HTTP call.
+ * @param exchangesApi The [ExchangesApi] instance for interacting with TBDex exchanges.
+ * @param callback The optional callback function to be executed after a successful close submission.
+ */
+@Suppress("TooGenericExceptionCaught", "MaxLineLength", "SwallowedException")
```
Comment (phoebe-lew) [inline httpserver/src/main/kotlin/tbdex/sdk/httpserver/handlers/SubmitMessage.kt:24]: Unrelated but I haven't experienced `TooGenericExceptionCaught` or `SwallowedException` being useful EVER

**Review by phoebe-lew — APPROVED** [review]
Thank you!!

**Pull request merged**